### PR TITLE
Add tests from all the issues analyzer fixes

### LINKED ISSUE #4432: WITH array and "No alias for non-trivial value in ARRAY JOIN"
```sql
SELECT version()

┌─version()─┐
│ 19.3.3    │
└───────────┘

-- ok:
SELECT x
FROM system.one 
ARRAY JOIN [1, 2] AS x

┌─x─┐
│ 1 │
│ 2 │
└───┘

-- ok:

SELECT 
    [1, 2] AS zz, 
    x
FROM system.one 
ARRAY JOIN zz AS x

┌─zz────┬─x─┐
│ [1,2] │ 1 │
│ [1,2] │ 2 │
└───────┴───┘

-- wrong:
WITH [1, 2] AS zz
SELECT x
FROM system.one 
ARRAY JOIN zz AS x

Received exception from server (version 19.3.3):
Code: 206. DB::Exception: Received from localhost:9000, ::1. DB::Exception: No alias for non-trivial value in ARRAY JOIN: x. 
```

### LINKED ISSUE #4567: CH has to throw errors about ambiguous columns.
create table t(a Int64, b Int64) engine = TinyLog;
create table s(a Int64, b Int64) engine = TinyLog;

select a from t left outer join s on (t.b=s.b);

CH: 0 rows in set. Elapsed: 0.001 sec.
PG: ERROR:  column reference "a" is ambiguous
mysql: ERROR 1052 (23000): Column 'a' in field list is ambiguous

select t.a from t left outer join s on (t.b=s.b) where a=1;

CH: 0 rows in set. Elapsed: 0.002 sec.
PG: ERROR:  column reference "a" is ambiguous
mysql: ERROR 1052 (23000): Column 'a' in where clause is ambiguous

### LINKED ISSUE #6571: Join between distributed (locally defined) and sharded table (not locally defined)
ORDERS is a distributed table defined over many ORDERS_shard tables.
LINEITEM is a distributed table defined over many LINEITEM_shard tables.

This query

```
select 
	O_ORDERPRIORITY,
	count(*) as order_count
from ORDERS  JOIN (
	select L_ORDERKEY
	from
	LINEITEM_shard
	group by L_ORDERKEY
	having any(L_COMMITDATE < L_RECEIPTDATE)
) on O_ORDERKEY=L_ORDERKEY
group by O_ORDERPRIORITY
order by O_ORDERPRIORITY
limit 1;
```

works by performing local joins between ORDERS_shard and LINEITEM_shard but **requires LINEITEM_shard to be defined in the node that receives the initial query (just for validating the query)**. 
If not we get: 

```
Code: 60. DB::Exception: Received from ...:9000, .... DB::Exception: Table LINEITEM_shard doesn't exist.. 

0. /usr/bin/clickhouse-server(StackTrace::StackTrace()+0x16) [0x72865f6]
1. /usr/bin/clickhouse-server(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x22) [0x39a90f2]
2. /usr/bin/clickhouse-server(DB::Context::getTableImpl(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::Exception*) const+0x382) [0x668b0e2]
3. /usr/bin/clickhouse-server(DB::Context::getTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const+0x73) [0x668b2d3]
4. /usr/bin/clickhouse-server(DB::InterpreterSelectQuery::InterpreterSelectQuery(std::shared_ptr<DB::IAST> const&, DB::Context const&, std::shared_ptr<DB::IBlockInputStream> const&, std::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x10b0) [0x66e4c50]
5. /usr/bin/clickhouse-server(DB::InterpreterSelectQuery::InterpreterSelectQuery(std::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x4a) [0x66e4e9a]
6. /usr/bin/clickhouse-server(DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x76f) [0x66f1e9f]
7. /usr/bin/clickhouse-server(DB::InterpreterSelectWithUnionQuery::getSampleBlock(std::shared_ptr<DB::IAST> const&, DB::Context const&)+0x149) [0x66f2f79]
8. /usr/bin/clickhouse-server(DB::getNamesAndTypeListFromTableExpression(DB::ASTTableExpression const&, DB::Context const&)+0x74) [0x6c8c934]
9. /usr/bin/clickhouse-server(DB::SyntaxAnalyzer::analyze(std::shared_ptr<DB::IAST>&, DB::NamesAndTypesList const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::shared_ptr<DB::IStorage>) const+0x952) [0x67ad5a2]
10. /usr/bin/clickhouse-server(DB::InterpreterSelectQuery::InterpreterSelectQuery(std::shared_ptr<DB::IAST> const&, DB::Context const&, std::shared_ptr<DB::IBlockInputStream> const&, std::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x49e) [0x66e403e]
11. /usr/bin/clickhouse-server(DB::InterpreterSelectQuery::InterpreterSelectQuery(std::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x4a) [0x66e4e9a]12. /usr/bin/clickhouse-server(DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x76f) [0x66f1e9f]
13. /usr/bin/clickhouse-server(DB::InterpreterFactory::get(std::shared_ptr<DB::IAST>&, DB::Context&, DB::QueryProcessingStage::Enum)+0x164) [0x66c9464]
14. /usr/bin/clickhouse-server() [0x67e2bc9]
15. /usr/bin/clickhouse-server(DB::executeQuery(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool)+0x74) [0x67e2e44]
16. /usr/bin/clickhouse-server(DB::TCPHandler::runImpl()+0x730) [0x39b49f0]
17. /usr/bin/clickhouse-server(DB::TCPHandler::run()+0x2b) [0x39b503b]
18. /usr/bin/clickhouse-server(Poco::Net::TCPServerConnection::start()+0xf) [0x73956bf]
19. /usr/bin/clickhouse-server(Poco::Net::TCPServerDispatcher::run()+0xe9) [0x7395df9]
20. /usr/bin/clickhouse-server(Poco::PooledThread::run()+0x81) [0x7446a71]
21. /usr/bin/clickhouse-server(Poco::ThreadImpl::runnableEntry(void*)+0x38) [0x7442e78]
22. /usr/bin/clickhouse-server() [0xb1a35df]
23. /lib64/libpthread.so.0(+0x7dd5) [0x7f18a79eedd5]
24. /lib64/libc.so.6(clone+0x6d) [0x7f18a7009ead]
```

If the nodes that process the query don't store data (ORDERS is a Distributed query defined over a remote cluster) we need to setup an empty (don't use the data, just the definition) LINEITEM_shard to be able to execute the query.

**I'd expect that validation to be done on the remote cluster.**

### LINKED ISSUE #7520: GROUP BY using WITH columns alias don't work if we call the column using the column name on SELECT
**Describe the bug or unexpected behaviour**

If we do a `GROUP BY` using an alias column on the `WITH`, it consider the following column as not part of the `ORDER BY`

**How to reproduce**
Clickhouse server v19.14.3
```
CREATE TABLE test (`a` UInt32, `b` UInt32) ENGINE = Memory;

WITH
    a as key
SELECT
    a as k1,
    sum(b) as k2
FROM
    test
GROUP BY
    key
ORDER BY k1, k2;

Code: 215. DB::Exception: Received from 127.0.0.1:10000. DB::Exception: Column a is not under aggregate function and not in GROUP BY.. 
```

**Expected behavior**
This query should be executable as the alias `key` and the column `a` share the same expression.

### LINKED ISSUE #8259: Unable to use CTE scalar as numbers() argument
This is a simplification of what I'm trying to achieve:

```
with
    (select 25) as something
select *, something
from numbers(toUInt64(something));
```

It fails with error: `Code: 47, e.displayText() = DB::Exception: Missing columns: 'something' while processing query: 'toUInt64(something)', required columns: 'something', source columns: '_dummy' (version 19.17.5.18 (official build))`

Replace last line with a fixed value, and the query works fine. Is it not possible to use a CTE value as  input to numbers()? Any workaround?

thanks!

### LINKED ISSUE #9233: ARRAY JOIN allows non-existing columns (UB)
Y -- does not exists, but ARRAY JOIN does not check it.

```
SELECT *
FROM
(
    SELECT
        [1, 2, 3] AS x,
        [4, 5, 6] AS y
)
ARRAY JOIN
    x,
    Y

┌─x─┬─y───────┐
│ 1 │ [4,5,6] │
│ 2 │ [4,5,6] │
│ 3 │ [4,5,6] │
└───┴─────────┘
```

### LINKED ISSUE #9954: Not found column in block
```
:) select 1 value, * from (select 1 key) l left join (select 2 key, 1 value) r using key;

Received exception from server (version 20.4.1):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column value in block. There are only columns: key, 1. Stack trace:
```

The query looks valid though.

### LINKED ISSUE #10276: Unexpected Error of alias which redefines column name in join
Alias of the redefined column name in multi join

**How to reproduce**
version 19.17.6.36 
Normal Case
```
SELECT 
    sum(x.n) as n1,
    sum(z.n) as n2
FROM
(
    SELECT 1000 AS n,1 as id
) AS x 
join (select 10000 as n,1 as id) as y 
on x.id = y.id 
left join (select 100000 as n,1 as id) as z 
on x.id = z.id 
```
Result

n1 | n2 
-- | -- 
1000 | 100000

Unexpected Case
```
SELECT 
    sum(x.n) as n,
    sum(z.n) as n2
FROM
(
    SELECT 1000 AS n,1 as id
) AS x 
join (select 10000 as n,1 as id) as y 
on x.id = y.id 
left join (select 100000 as n,1 as id) as z 
on x.id = z.id 
```
Result
`Code: 10, e.displayText() = DB::Exception: Not found column n in block. There are only columns: --x.n, --x.id (version 19.17.6.36 (official build))`

I also met another Error in the SQL with the same struct

`Code: 47, e.displayText() = DB::Exception: Unknown identifier: booked_guaranteed_narrower_or_equal (version 19.17.6.36 (official build))
`

So how to solve this problem?

### LINKED ISSUE #11068: Using alias in simple view
I try to make a view... for the moment i have this request : 

```
drop table if exists computed_datum_hours;
create view computed_datum_hours as
select
    toStartOfHour(b.datetime_value) as datetime_desc,
    b.computed_id,
    b.computed_kind,
    b.computed_type,
    sum(b.value) as value
from computed_datum as b
group by
    b.computed_id,
    b.computed_kind,
    b.computed_type,
    toStartOfHour(b.datetime_value)
;
```
but clickhouse say :
`Code: 47, e.displayText() = DB::Exception: Missing columns: 'b.value' 'b.datetime_value' 'b.computed_type' 'b.computed_kind' 'b.computed_id' while processing query: 'SELECT toStartOfHour(b.datetime_value) AS datetime_desc, b.computed_id, b.computed_kind, b.computed_type, sum(b.value) AS value FROM api_client.computed_datum GROUP BY b.computed_id, b.computed_kind, b.computed_type, toStartOfHour(b.datetime_value)', required columns: 'b.computed_id' 'b.computed_kind' 'b.computed_type' 'b.datetime_value' 'b.value', source columns: 'flags' 'computed_type' 'datetime_value' 'uuid' 'value' 'computed_id' 'insert' 'computed_kind' 'EventDateTime' (version 20.4.2.9 (official build))`

this is a bug or alias is different from mariadb ? 
( PS, if i remove every alias to b , the view work )

### LINKED ISSUE #11757: cross join + array join: not found column in block
```
~3~ :) select * from (select [1, 2] a) aa cross join (select [3, 4] b) bb array join aa.a, bb.b;

Received exception from server (version 20.5.1):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column b in block. There are only columns: a. Stack trace:
```

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add tests for all issues which are no longer relevant with Analyzer being enabled by default.
Closes: https://github.com/ClickHouse/ClickHouse/issues/55794 Closes: https://github.com/ClickHouse/ClickHouse/issues/49472 Closes: https://github.com/ClickHouse/ClickHouse/issues/44414 Closes: https://github.com/ClickHouse/ClickHouse/issues/13843 Closes: https://github.com/ClickHouse/ClickHouse/issues/55803 Closes: https://github.com/ClickHouse/ClickHouse/issues/48308 Closes: https://github.com/ClickHouse/ClickHouse/issues/45535 Closes: https://github.com/ClickHouse/ClickHouse/issues/44365 Closes: https://github.com/ClickHouse/ClickHouse/issues/44153 Closes: https://github.com/ClickHouse/ClickHouse/issues/42399 Closes: https://github.com/ClickHouse/ClickHouse/issues/27115 Closes: https://github.com/ClickHouse/ClickHouse/issues/23162 Closes: https://github.com/ClickHouse/ClickHouse/issues/15395 Closes: https://github.com/ClickHouse/ClickHouse/issues/15411 Closes: https://github.com/ClickHouse/ClickHouse/issues/14978 Closes: https://github.com/ClickHouse/ClickHouse/issues/17319 Closes: https://github.com/ClickHouse/ClickHouse/issues/11813 Closes: https://github.com/ClickHouse/ClickHouse/issues/13210 Closes: https://github.com/ClickHouse/ClickHouse/issues/23053 Closes: https://github.com/ClickHouse/ClickHouse/issues/37729 Closes: https://github.com/ClickHouse/ClickHouse/issues/32639 Closes: https://github.com/ClickHouse/ClickHouse/issues/9954 Closes: https://github.com/ClickHouse/ClickHouse/issues/41964 Closes: https://github.com/ClickHouse/ClickHouse/issues/54317 Closes: https://github.com/ClickHouse/ClickHouse/issues/7520 Closes: https://github.com/ClickHouse/ClickHouse/issues/36973 Closes: https://github.com/ClickHouse/ClickHouse/issues/40955 Closes: https://github.com/ClickHouse/ClickHouse/issues/19687 Closes: https://github.com/ClickHouse/ClickHouse/issues/23104 Closes: https://github.com/ClickHouse/ClickHouse/issues/21584 Closes: https://github.com/ClickHouse/ClickHouse/issues/23344 Closes: https://github.com/ClickHouse/ClickHouse/issues/22627 Closes: https://github.com/ClickHouse/ClickHouse/issues/10276 Closes: https://github.com/ClickHouse/ClickHouse/issues/19687 Closes: https://github.com/ClickHouse/ClickHouse/issues/4567 Closes: https://github.com/ClickHouse/ClickHouse/issues/17710 Closes: https://github.com/ClickHouse/ClickHouse/issues/11068 Closes: https://github.com/ClickHouse/ClickHouse/issues/24395 Closes: https://github.com/ClickHouse/ClickHouse/issues/23416 Closes: https://github.com/ClickHouse/ClickHouse/issues/23162 Closes: https://github.com/ClickHouse/ClickHouse/issues/25655 Closes: https://github.com/ClickHouse/ClickHouse/issues/11757 Closes: https://github.com/ClickHouse/ClickHouse/issues/6571 Closes: https://github.com/ClickHouse/ClickHouse/issues/4432 Closes: https://github.com/ClickHouse/ClickHouse/issues/8259 Closes: https://github.com/ClickHouse/ClickHouse/issues/9233 Closes: https://github.com/ClickHouse/ClickHouse/issues/14699 Closes: https://github.com/ClickHouse/ClickHouse/issues/27068 Closes: https://github.com/ClickHouse/ClickHouse/issues/28687 Closes: https://github.com/ClickHouse/ClickHouse/issues/28777 Closes: https://github.com/ClickHouse/ClickHouse/issues/29734 Closes: https://github.com/ClickHouse/ClickHouse/issues/61238 Closes: https://github.com/ClickHouse/ClickHouse/issues/33825 Closes: https://github.com/ClickHouse/ClickHouse/issues/35608 Closes: https://github.com/ClickHouse/ClickHouse/issues/29838 Closes: https://github.com/ClickHouse/ClickHouse/issues/35652 Closes: https://github.com/ClickHouse/ClickHouse/issues/36189 Closes: https://github.com/ClickHouse/ClickHouse/issues/39634 Closes: https://github.com/ClickHouse/ClickHouse/issues/47432 Closes: https://github.com/ClickHouse/ClickHouse/issues/54910 Closes: https://github.com/ClickHouse/ClickHouse/issues/57321 Closes: https://github.com/ClickHouse/ClickHouse/issues/59154 Closes: https://github.com/ClickHouse/ClickHouse/issues/61014 Closes: https://github.com/ClickHouse/ClickHouse/issues/61950 Closes: https://github.com/ClickHouse/ClickHouse/issues/55647 Closes: https://github.com/ClickHouse/ClickHouse/issues/61947

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit ed9ee5ab4cfa56cd615024d20f1d7d1c31b88be3 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62185/ed9ee5ab4cfa56cd615024d20f1d7d1c31b88be3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by fm4v** [comment]
Closes: https://github.com/ClickHouse/ClickHouse/issues/23162 (Test added: 02694_wrong_identifier_shouldnt_be_accepted, confirmed)
Closes: https://github.com/ClickHouse/ClickHouse/issues/15395 (Test added: 03040_array_sum_and_join)

File: tests/queries/0_stateless/03034_normalized_ast.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- https://github.com/ClickHouse/ClickHouse/issues/49472
+SELECT
+    concat(database, table) AS name,
+    count()
+FROM clusterAllReplicas(default, system.tables)
```
Comment (Algunenano) [inline tests/queries/0_stateless/03034_normalized_ast.sql:5]: The `default` cluster doesn't exist anymore for tests ( https://github.com/ClickHouse/ClickHouse/pull/61396). You can use `test_shard_localhost` instead

Comment (nikitamikhaylov) [reply tests/queries/0_stateless/03034_normalized_ast.sql:5]: Thanks!

Comment (nikitamikhaylov) [inline tests/queries/0_stateless/03034_normalized_ast.sql:5]:
```suggestion
FROM clusterAllReplicas(test_shard_localhost, system.tables)
```

File: tests/queries/0_stateless/03035_alias_column_bug_distributed.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+(
+    `src` String,
+    `theAlias` String ALIAS trimBoth(src)
+)
+ENGINE = MergeTree()
+ORDER BY src;
+
+CREATE TABLE alias_bug_dist
+AS alias_bug
+ENGINE = Distributed('default', currentDatabase(), 'alias_bug', rand());
```
Comment (nikitamikhaylov) [inline tests/queries/0_stateless/03035_alias_column_bug_distributed.sql:14]:
```suggestion
ENGINE = Distributed('test_shard_localhost', currentDatabase(), 'alias_bug', rand());
```

File: tests/queries/0_stateless/03035_alias_column_bug_distributed.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    `s` String,
+    `src` String,
+    `theAlias` String ALIAS trimBoth(src)
+)
+ENGINE = MergeTree()
+ORDER BY src;
+
+CREATE TABLE alias_bug_dist
+AS alias_bug
+ENGINE = Distributed('default', currentDatabase(), 'alias_bug', rand());
```
Comment (nikitamikhaylov) [inline tests/queries/0_stateless/03035_alias_column_bug_distributed.sql:37]:
```suggestion
ENGINE = Distributed('test_shard_localhost', currentDatabase(), 'alias_bug', rand());
```

**Comment by nikitamikhaylov** [comment]
I cannot believe this... 
image
Let's trigger the CI one more time.

**Comment by fm4v** [comment]
This issue https://github.com/ClickHouse/ClickHouse/issues/19687 doesn't have reproducible example but this PR have similar tests with JOIN from distributed tables.
Close: https://github.com/ClickHouse/ClickHouse/issues/19687

**Comment by fm4v** [comment]
Looks like analyzer problem but doesn't have reproducible example: https://github.com/ClickHouse/ClickHouse/issues/53141

**Comment by qoega** [comment]
best. pr. ever.

**Comment by nikitamikhaylov** [comment]
Waiting for the fast test only.

**Pull request merged**